File: Server/Server/UpdateMessage.cpp
#include "UpdateMessage.h"

UpdateMessage::UpdateMessage() {
	this->type = UpdateType::WND_FOCUSED;
	this->wndId = 0;
	this->wndName = "";
	this->procName = "";
	this->wndIcon = "";
}

UpdateMessage::UpdateMessage(UpdateType type, int wndId, std::string wndName, std::string procName, std::string wndIcon) {
	this->type = type;
	this->wndId = wndId;
	this->wndName = wndName;
	this->procName = procName;
	this->wndIcon = wndIcon;
}

void UpdateMessage::serialize(cereal::JSONOutputArchive& archive) {
	archive(CEREAL_NVP(type), CEREAL_NVP(wndId), CEREAL_NVP(wndName), CEREAL_NVP(procName), CEREAL_NVP(wndIcon));
}

void UpdateMessage::deserialize(cereal::JSONInputArchive& archive) {
	archive(type, wndId, wndName, procName, wndIcon);
}

File: Server/UpdateMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

enum UpdateType { WND_CREATED = 0, WND_FOCUSED = 1, WND_DESTROYED = 2 };
enum KeyCombTarget { FOCUSED_WND = 0, SPECIFIC_WND = 1 };

namespace ClientTest {
    class UpdateMessage {
        public UpdateType type;
        public int wndId;
        public string wndName;
        public string wndIcon;
    }
}
